### PR TITLE
pds-api #91 : do not support html/xml and force JSON

### PR DESCRIPTION
**Pull request by al-niessner**
**Summary***
Removed the XML to HTML/XML converter.

**Test Data and/or Report**
Visited `http://localhost:8080/products` with results attached as image. It generates an error deep in Apache and SpringBoard tools:

```
2021-08-31 18:06:38.557 DEBUG 53 --- [0.3-8080-exec-1] o.s.w.s.m.m.a.HttpEntityMethodProcessor  : Using 'application/xml;q=0.9', given [text/html, application/xhtml+xml, image/webp, application/xml;q=0.9, */*;q=0.8] and supported [application/json, application/xml]
2021-08-31 18:06:38.559  WARN 53 --- [0.3-8080-exec-1] .w.s.m.s.DefaultHandlerExceptionResolver : Resolved [org.springframework.http.converter.HttpMessageNotWritableException: No converter for [class gov.nasa.pds.model.Products] with preset Content-Type 'null']
2021-08-31 18:06:38.559 DEBUG 53 --- [0.3-8080-exec-1] o.s.web.servlet.DispatcherServlet        : Completed 500 INTERNAL_SERVER_ERROR
2021-08-31 18:06:38.561 DEBUG 53 --- [0.3-8080-exec-1] o.a.c.c.C.[Tomcat].[localhost]           : Processing ErrorPage[errorCode=0, location=/error]
2021-08-31 18:06:38.564 DEBUG 53 --- [0.3-8080-exec-1] o.s.web.servlet.DispatcherServlet        : "ERROR" dispatch for GET "/error", parameters={}
2021-08-31 18:06:38.564 DEBUG 53 --- [0.3-8080-exec-1] pertySourcedRequestMappingHandlerMapping : looking up handler for path: /error
2021-08-31 18:06:38.566 DEBUG 53 --- [0.3-8080-exec-1] s.w.s.m.m.a.RequestMappingHandlerMapping : Mapped to org.springframework.boot.autoconfigure.web.servlet.error.BasicErrorController#errorHtml(HttpServletRequest, HttpServletResponse)
2021-08-31 18:06:38.578 DEBUG 53 --- [0.3-8080-exec-1] o.s.web.servlet.DispatcherServlet        : Exiting from "ERROR" dispatch, status 500
```

It looks as though a different and more informative error could be generated but this should be a different ticket because this error will also cover application/pickle as well as application/xml etc.

On firefox, the error looks like this (not informative):

error_msg

**Related Issues**
None

### CODE REVIEW

**Comment by al-niessner** [comment]
@tloubrieu-jpl @jordanpadams 

Please review this pull request (initial block) for outstanding questions.

File: src/main/java/gov/nasa/pds/api/engineering/configuration/WebMVCConfig.java [hunk 1]
```diff
@@ -56,52 +37,41 @@ public void addResourceHandlers(ResourceHandlerRegistry registry) {
...
+	}
 
-  
-  
-  
+	/**
+	 * Setup a simple strategy: use all the defaults and return XML by default when not sure. 
+	 */
+	@SuppressWarnings("deprecation")
+	public void configureContentNegotiation(ContentNegotiationConfigurer configurer) {
```
[the remark this answers is not in the document]
Comment (al-niessner) [reply src/main/java/gov/nasa/pds/api/engineering/configuration/WebMVCConfig.java:51]: fixed

**Review by tloubrieu-jpl — APPROVED** [review]
This pull request is ok for now.

For clearer behavior, I would remove the XML support from the openAPI definition and from the Json serializer.

**Review by tloubrieu-jpl — CHANGES_REQUESTED** [review]
Sorry, given the error you are showing above on the HTML error, @al-niessner, I will not merge the pull request now.

This error is because you made the Json serializer accept XML or HTML format requests therefore firefox/chrome/... believe this is the format it is getting whereas it is JSON.

To remove this error, you must remove the application/xml support from the swagger definition of the API and remove it from the Json serializer as well. We will also need to remove these lines from the controllers:
```

|| accept.contains("text/html")
|| accept.contains("application/xml")


```


Since I rebase this branch on main, @al-niessner you should pull from origin to get the lqtest version

Thanks

**Comment by al-niessner** [comment]
Not visible on this pull request are changes taking place in [pds-api-javalib](https://github.com/NASA-PDS/pds-api-javalib/tree/swaggerhub-pds_federated_api-pds-api-91) which is the pushed version from [swagger](https://app.swaggerhub.com/apis/PDS_APIs/pds_federated_api/pds-api-91) that removes all references to application/xml -- but they are only commented out since there is persistence remarks of supporting xml in the future.

**Comment by al-niessner** [comment]
@jordanpadams @tloubrieu-jpl 

Not happy with error one gets when requesting a format not supported. It is a general error handler that should be specific to application/? is not supported so that the end user clearly knows what to change.

If you agree, then please open a new ticket.

**Comment by jordanpadams** [comment]
> Not visible on this pull request are changes taking place in [pds-api-javalib](https://github.com/NASA-PDS/pds-api-javalib/tree/swaggerhub-pds_federated_api-pds-api-91) which is the pushed version from [swagger](https://app.swaggerhub.com/apis/PDS_APIs/pds_federated_api/pds-api-91) that removes all references to application/xml -- but they are only commented out since there is persistence remarks of supporting xml in the future.

I just merged these changes into main and pushed to pds-api-javalib: https://github.com/NASA-PDS/pds-api-javalib/commit/8bc11a75ca400943e582a4f25cb425118b4b9452

**Review by tloubrieu-jpl — APPROVED** [review]

**Comment by tloubrieu-jpl** [comment]
I agree with you @al-niessner the error message should be improved for non accepted format as stated in https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/406

I will merge the pull request and create a new ticket to comply with the recommendation on the previous web page.

**Pull request merged**